app/src/main/java/org/mozilla/FirefoxLite/firefoxdownloader/SampleService.java

```diff
@@ -0,0 +1,249 @@
+package org.mozilla.FirefoxLite.firefoxdownloader;
+
+import android.app.Notification;
+import android.app.Service;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.SharedPreferences;
+import android.graphics.Color;
+import android.os.IBinder;
+import android.preference.PreferenceManager;
+import android.util.JsonReader;
+import android.util.Log;
+import android.widget.Toast;
+import androidx.annotation.Nullable;
+import androidx.core.app.NotificationCompat;
+import androidx.lifecycle.LifecycleOwner;
+import androidx.lifecycle.LifecycleService;
+
+import static com.sample.NotificationHelper.channel1ID;
+
+public class SampleService extends LifecycleService {
+    final static String MY_ACTION = "MY_ACTION";
+
+    final String URL1 = "http://mattmahoney.net/dc/enwik8.pmd";
+    private static String dirPath;
+    int downloadIdOne;
+    NotificationHelper mNotificationHelper;
+    MainActivity mainActivity;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        mainActivity=new MainActivity();
+        mNotificationHelper = new NotificationHelper(this);
+        dirPath = Utils.getRootDirPath(getApplicationContext());
+
+    }
+
+    private void sendOnchannel1(String title, String message) {
+        // NotificationCompat.Builder nb = mNotificationHelper.getChannel1Notification(title, message);
+        //  mNotificationHelper.getManager().notify(1, nb.build());
+
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        Log.d("nnull","unn");
+        String downloadstatus = intent.getStringExtra("downloadstatus");
+        String schedulestatus = intent.getStringExtra("schedulestatus");
+        String removeschedulling = intent.getStringExtra("scheduleoption");
+        if(removeschedulling!=null){
+            if(removeschedulling.equals("removeschedulling")){
+                MozzDownloader.cancel(downloadIdOne);
+                Notification notification=new NotificationCompat.Builder(getApplicationContext(),channel1ID)
+                        .setContentTitle("mandy.pmd").setContentText("Schedule Terminated").setSmallIcon(R.drawable.ic_launcher_background).build();
```

Comment:
where do you get this title? is it always pmd?

Reply:
@This is the extension of the firlename like cat.png or .jpeg ,no right now it hard coded according to url.Can you suggest any other way for this?